src/main/java/com/rusticisoftware/tincan/RemoteLRS.java

```diff
@@ -223,19 +220,113 @@ public void onHeaders(Response webResp) {
+                        webReq.header("Content-Type", "application/octet-stream");
+                    }
+
+                    webReq.content(content).send(listener);
+
+                    if (req.getContent() != null) {
+                        output.write(req.getContent());
+                    }
+
+                    output.close();
+                }
+                else {
+
+                    MultiPartOutputStream multiout = new MultiPartOutputStream(output);
+
+                    webReq.header("Content-Type", "multipart/mixed; boundary=" + multiout.getBoundary());
+                    webReq.content(content).send(listener);
+
+                    if (req.getContentType() != null) {
+                        multiout.startPart(req.getContentType());
+                    } else {
+                        multiout.startPart("application/octet-stream");
+                    }
+
+                    if (req.getContent() != null) {
+                        multiout.write(req.getContent());
+                    }
+
+                    for (HTTPPart part : req.getPartList()) {
+                        multiout.startPart(part.getContentType(), new String[]{
+                                "Content-Transfer-Encoding: binary",
+                                "X-Experience-API-Hash: " + part.getSha2()});
+                        multiout.write(part.getContent());
+                    }
+                    multiout.close();
+                }
             }
-            else {
-                webReq.header("Content-Type", "application/octet-stream");
+
+            Response httpResponse = listener.get(5, TimeUnit.SECONDS);
+
+            response.setStatus(httpResponse.getStatus());
+            response.setStatusMsg(httpResponse.getReason());
+            for (HttpField header : httpResponse.getHeaders()) {
+                response.setHeader(header.getName(), header.getValue());
             }
 
-            if (req.getContent() != null) {
-                webReq.content(new BytesContentProvider(req.getContent()));
+
+
+            // Use try-with-resources to close input stream.
+            try (InputStream responseContent = listener.getInputStream())
+            {
+
+                if (response.getContentType() != null && response.getContentType().contains("multipart/mixed")) {
+
+                    MultipartParser responseHandler = new MultipartParser(responseContent);
+
+                    // We need to get the first part that contains the statements and parse them
+                    responseHandler.nextPart();
+
+                    ArrayList<Statement> statements = new ArrayList<Statement>();
+
+                    if (responseHandler.getHeaders().get("Content-Type").contains("application/json")) {
+                        JsonNode statementsNode = (new StringOfJSON(new String(responseHandler.getContent())).toJSONNode());
+                        if (! (statementsNode.findPath("statements").isMissingNode())) {
```

Comment:
Double negation feels more complicated than necessary here, isn't it sufficient to check to see if it is a missing node in which case we assume it is a single statement, otherwise we assume it is an array of statements?